lib/segment/src/vector_storage/query/reco_query.rs

```diff
@@ -125,18 +128,30 @@ mod test {
         }
     }
 
+    /// Relaxes the comparison of floats to allow for a some difference in units of least precision
+    fn float_cmp(a: f32, b: f32) -> Ordering {
+        if approx_eq!(f32, a, b, ulps = 80) {
```

Comment:
```suggestion
        if (a - b).abs() < 0.0001 {
```

Reply:
IMO (after just learning about them) units of least precision make more sense, since they are less sensitive to the magnitude of the floats.

Reply:
1. I have to google about ulps to understand this
2. `ulps = 80` is as arbitrary as `0.0001`
3. This small function doesn't justify including another crate

Reply:
Ok

Reply:
I had to relax it to < 0.001 to make the inputs `a = 62.9219, b = 62.921734` pass